### ISSUE TITLE
Audiobooks - Add note regarding ZIP compression method 'store' for media files

### ISSUE THREAD

**Issue by larscwallin** [w3c/wpub#460]
The ZIP format allows individual files to specify compression algorithm. The default options are Store and Deflate. Store leaves the file uncompressed while Deflate compresses the file. 
Many ZIP libraries will compress media files such as audio and video using the Deflate algorithm. When compressed, the Deflate algorithm will affect the media files in a way that makes "seeking" into them directly from the archive, without first extracting them, impossible. The result is that you always need to start from the beginning of the stream when reading from the file, even if you want to start playback 10 seconds from the end.

I would like to suggest that we add a section to the Audiobooks spec regarding compression that recommends, or even requires media files to be 'stored' rather than 'deflated'. This would be a highly logical addition to the format since it should be optimized for streaming I think.

The drawbacks are few. As far as file size is concerned it makes little difference as most media formats already are compressed (mp4, mp3, etc.)

**Comment by llemeurfr** [comment]
@larscwallin you're right and this is what is already our practice for encrypted media files in the LCP case. Deflating a binary resource is not useful at all (size wise) and disallows "streaming" (byte range access, seek).  

Your remark is directed to the Lightweight Packaging Format, I'll propose to require binary media files to be "stored" in the zip archive.

**Comment by iherman** [comment]
While I agree in principle, there may be a usage issue. The way the LPF is done today is such that the author has to provide some specific types of data but then any zip software can be used to zip the content and get it shipped. E.g., on the Mac, I can use the built-in zip facility in the finder to do so. If we set this as a requirement, we get to the same situation as for EPUB, whereby one needs some specific scripts (or similar) that is parameterized accordingly, and that may be a problem for some.

(One may say that most of the authoring may be done via specialized tools eventually, so that problem is not real. I do no know...)

**Comment by llemeurfr** [comment]
Good remark @iherman. Therefore a SHOULD is a way to keep the packaging process flexible but push packaging tools to do the right think. The load will be on the "streaming" tools, for sure; but even if a MUST is written, some implementers will not read it and streaming tools will have to deal with the issue.

**Comment by laudrain** [comment]
@iherman for EPUB, the EPUBCheck tool that everyone uses for conformance of the final structure and content, has a save mode that does the zip epub file from a non compressed folder. So it is a straightforward process.

**Comment by danielweck** [comment]
My developer feedback: in the context of Readium, "seek" and "tell" on deflated binary streams needed to be implemented in order to enable the use-case of accessing arbitrary ranges of bytes (e.g. respond to HTTP partial requests for video/audio resources).
There is undoubtedly a processing penalty when the stream "cursor" needs to be rewinded or forwarded by great distances (which is implemented under the hood by reading bytes / filling temporary buffers). However in practice this has not been problematic, in the sense that the user experience / perceived performance did not degrade much. By "in practice", I mean with typical EPUBs that contain video media, as well as long-ish Media Overlays (synchronized text/audio). This is in large part due to the fact that the linear nature of an average reading/listening/watching experience does not frequently require translating the stream "cursor" by great distances.

That being said, I agree with other comments that we should encourage / recommend *storing* (instead of *deflating*) resources that are intrinsically "streamable", and for assets that are already binary-encoded with compression strategies (images?).
